synedrion/src/tools/secret.rs

```diff
@@ -39,22 +73,349 @@ impl<T: Zeroize> Debug for Secret<T> {
     }
 }
 
-impl<T: Zeroize> From<SecretBox<T>> for Secret<T> {
-    fn from(value: SecretBox<T>) -> Self {
-        Self(value)
+impl<T> Secret<T>
+where
+    T: Zeroize + Clone + HasWide,
+    T::Wide: Zeroize,
+{
+    pub fn to_wide(&self) -> Secret<<T as HasWide>::Wide> {
+        Secret::init_with(|| {
+            let mut value = self.expose_secret().clone();
+            let result = value.to_wide();
+            value.zeroize();
+            result
+        })
+    }
+
+    pub fn mul_wide(&self, rhs: &T) -> Secret<T::Wide> {
+        Secret::init_with(|| {
+            let mut value = self.expose_secret().clone();
```

Comment:
So this is `mut` so that you can `zeroize` it yeah? I wonder if you can get rid of the `clone` and instead take ownership of the secret by `mem::swap`-ing in a `T::ZERO`?
The method takes `&self` which implies that the `Secret` can still be used afterwards, but I wonder how often we actually do that. Like, would a `mul_wide(self)` actually be a problem for us?

Reply:
We do take a reference here at least in some places where it comes from the caller, e.g. in `DecProof::new()`

Reply:
The explicit zeroization was removed